metrics/ci_integrations/test/source/jenkins/adapter/jenkins_source_client_adapter_test.dart

```diff
@@ -222,12 +226,55 @@ void main() {
 
         whenFetchBuilds().thenAnswer(responses.fetchBuilds);
 
-        final result = adapter.fetchBuilds(
+        final result = await adapter.fetchBuilds(
+          jobName,
+          fetchLimit,
+        );
+
+        expect(result, equals(expected));
+      },
+    );
+
+    test(
+      ".fetchBuilds() maps running builds to in-progress builds",
+      () async {
+        final jenkinsBuilds = [
+          createJenkinsBuild(buildNumber: 1, building: true),
+          createJenkinsBuild(buildNumber: 2, building: true),
+        ];
+
+        responses.addBuilds(jenkinsBuilds);
+
+        whenFetchBuilds().thenAnswer(responses.fetchBuilds);
+
+        final result = await adapter.fetchBuilds(
+          jobName,
+          fetchLimit,
+        );
+
+        expect(result, everyElement(isInProgressBuild));
+      },
+    );
+
+    test(
+      ".fetchBuilds() maps running builds to build data with null duration",
```

Comment:
It seems like the `isInProgressBuild` predicate in the previous test case checks the duration to be `null`. Not sure this test case makes sense. I suppose we should either remove the duration checking from the predicate or get rid of this and similar test cases.